src/app/services/technical-record/technical-record.service.ts

```diff
@@ -257,4 +260,48 @@ export class TechnicalRecordService {
       }),
     );
   }
+
+  hasPsvGrossAxleChanged(changes: Partial<TechRecordGETPSV>): boolean {
+    return [
+      changes.techRecord_grossKerbWeight,
+      changes.techRecord_grossDesignWeight,
+      changes.techRecord_grossLadenWeight,
+      changes.techRecord_grossGbWeight,
+    ].some(Boolean);
+  }
+
+  hasHgvGrossAxleChanged(changes: Partial<TechRecordGETHGV>): boolean {
+    return [changes.techRecord_grossEecWeight, changes.techRecord_grossDesignWeight, changes.techRecord_grossGbWeight].some(Boolean);
+  }
+
+  hasTrlGrossAxleChanged(changes: Partial<TechRecordGETTRL>): boolean {
+    return [changes.techRecord_grossEecWeight, changes.techRecord_grossDesignWeight, changes.techRecord_grossGbWeight].some(Boolean);
+  }
+
+  hasHgvTrainAxleChanged(changes: Partial<TechRecordGETHGV>): boolean {
+    return [changes.techRecord_trainDesignWeight, changes.techRecord_trainGbWeight, changes.techRecord_trainEecWeight].some(Boolean);
+  }
+
+  hasPsvTrainAxleChanged(changes: Partial<TechRecordGETPSV>): boolean {
+    return [changes.techRecord_trainDesignWeight, changes.techRecord_maxTrainGbWeight].some(Boolean);
+  }
+
+  hasMaxTrainAxleChanged(changes: Partial<TechRecordGETHGV>): boolean {
+    return [
+      changes.techRecord_maxTrainDesignWeight,
+      changes.techRecord_maxTrainEecWeight,
+      changes.techRecord_maxTrainGbWeight,
+    ].some(Boolean);
+  }
+
+  haveAxlesChanged(vehicleType: VehicleTypes, changes: Partial<TechRecordType<'get'>>) {
+    if (vehicleType === 'psv' && this.hasPsvGrossAxleChanged(changes as Partial<TechRecordGETPSV>)) return true;
```

Comment:
You could reduce the conditional complexity of this with some or || logic which would make checking the vehicle type more then once redundant e.g. if (vehicletype && (this || that ))